src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -53,10 +54,14 @@ public async Task Invoke(HttpContext context)
             var authorizeData = endpoint?.Metadata.GetOrderedMetadata<IAuthorizeData>() ?? Array.Empty<IAuthorizeData>();
             if (authorizeData.Count() == 0)
```

Comment:
Drop the `?? Array.Empty<IAuthorizeData>()`

And test for `if (authorizeData == null || authorizeData.Count() == 0)`

Rather than invoking Linq `.Count()` on an empty array?